### DISCUSSION TITLE
Unable to get all branches using remote set-branches origin '*' command

### DISCUSSION THREAD

**Discussion by Ekopalypse** [git-for-windows/git#3659] [Q&A] [answered]
- [x] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen) or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue matching what I'm seeing

### Setup

 - Which version of Git for Windows are you using? Is it 32-bit or 64-bit?
64bit
```
$ git --version --build-options

git version 2.35.0.windows.1
cpu: x86_64
built from commit: 853373155dce195a91e6e41a58c8b8e5c5a94652
sizeof-long: 4
sizeof-size_t: 8
shell-path: /bin/sh
feature: fsmonitor--daemon
```

 - Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?

```
$ cmd.exe /c ver

Microsoft Windows [Version 10.0.19042.1415] 64bit
```

 - What options did you set as part of the installation? Or did you choose the
   defaults?

```
# One of the following:
> type "C:\Program Files\Git\etc\install-options.txt"

Editor Option: VIM
Custom Editor Path:
Default Branch Option:
Path Option: Cmd
SSH Option: OpenSSH
Tortoise Option: false
CURL Option: OpenSSL
CRLF Option: CRLFAlways
Bash Terminal Option: MinTTY
Git Pull Behavior Option: Merge
Use Credential Manager: Enabled
Performance Tweaks FSCache: Enabled
Enable Symlinks: Disabled
Enable Pseudo Console Support: Disabled
Enable FSMonitor: Disabled
```

### Details

 - Which terminal/shell are you running Git from? e.g Bash/CMD/PowerShell/other

cmd

 - What commands did you run to trigger this issue? If you can provide a
   [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
   this will help us understand the issue.

```
On an existing repository which was cloned with --depth=1

git remote set-branches origin '*'
git pull
```
 - What did you expect to occur after running these commands?

No error and having all available branches

 - What actually happened instead?

Received the following message: 
```
Your configuration specifies to merge with the ref 'refs/heads/master'
from the remote, but no such ref was fetched.
```
only the master branch is available

Doing the same on Manjaro works.

mcve
```
C:\temp>git clone --depth=1 https://github.com/Ekopalypse/sdl
Cloning into 'sdl'...
remote: Enumerating objects: 99, done.
remote: Counting objects: 100% (99/99), done.
remote: Compressing objects: 100% (93/93), done.
Receiving objects: 100% (99/99), 766.87 KiB | 5.04 MiB/s, done.
Resolving deltas: 100% (2/2), done.42 (delta 0), pack-reused 0Resolving deltas:   0% (0/2)

C:\temp>cd sdl

C:\temp\sdl>git branch -a
* master
  remotes/origin/HEAD -> origin/master
  remotes/origin/master

C:\temp\sdl>git remote set-branches origin '*'

C:\temp\sdl>git pull
Your configuration specifies to merge with the ref 'refs/heads/master'
from the remote, but no such ref was fetched.

C:\temp\sdl>git ls-remote https://github.com/Ekopalypse/sdl
4cec9b9eb19b865bf25661a8e7ec1150ae867c72        HEAD
93801d866099f61997eb9f55b311eed541f976be        refs/heads/2.0.10
cce09b61210853b8dac6a6b6bf56a552eeffde0a        refs/heads/2.0.9
4cec9b9eb19b865bf25661a8e7ec1150ae867c72        refs/heads/master
b3282c68b2a75f6a3c374bb0581db272d2458835        refs/heads/powershell_automation
```

 - If the problem was occurring with a specific repository, can you provide the
   URL to that repository to help us with testing?

see mcve

**Comment by dscho** [comment, marked as answer]
> C:\temp\sdl>git remote set-branches origin '*'

If you have a look at the output of `git config -l --local`, you will notice that the single-quote characters are still there, which means that Git will then try to pull all branches whose names start and end with a single-quote character. That does not make sense, of course, as single-quote characters are not allowed in ref names.

The problem here is that CMD simply does not interpret the single-quote character as you probably expected. It does not treat it as a quoting anything at all.

You will want to use double-quote characters instead:

```cmd
C:\temp\sdl>git remote set-branches origin "*"
```

Make sure to verify the result via `git config -l --local` or via `git config --get-all remote.origin.fetch`.

**Comment by Ekopalypse** [comment]
Thank you very much. It is just as you said, with double quotes it works on Windows.
